This patch reorders the code in the EnterVT function in the tdfx driver
in an attempt to fix VT switch font corruption problems that are easily
reproduceable, and often reported.  The code is currently untested.  It
may also fix other VTswitch problems as well.

Mike A. Harris <mharris@redhat.com>

--- xc/programs/Xserver/hw/xfree86/drivers/tdfx/tdfx_driver.c.tdfx-fix-vtswitch-font-corruption	Sun Feb  3 08:40:10 2002
+++ xc/programs/Xserver/hw/xfree86/drivers/tdfx/tdfx_driver.c	Sun Feb  3 08:42:21 2002
@@ -2326,12 +2326,15 @@
   TDFXTRACE("TDFXLeaveVT start\n");
   pScrn = xf86Screens[scrnIndex];
   hwp=VGAHWPTR(pScrn);
-  TDFXRestore(pScrn);
-  vgaHWLock(hwp);
   pScreen = screenInfo.screens[scrnIndex];
+
   pTDFX = TDFXPTR(pScrn);
   pTDFX->sync(pScrn);
+
+  TDFXRestore(pScrn);
+  vgaHWLock(hwp);
   TDFXShutdownFifo(pScreen);
+
 #ifdef XF86DRI
   if (pTDFX->directRenderingEnabled) {
     DRILock(pScreen, 0);
